### DISCUSSION TITLE
2024/09/25/mruby-lite-intro/

### DISCUSSION THREAD

**Discussion by giscus** [katafrakt/website_comments#8] [Announcements]
# 2024/09/25/mruby-lite-intro/

I have always knew mruby exists but not much beyond that. In this post I share my journey in trying it out.

https://katafrakt.me/2024/09/25/mruby-lite-intro/

**Comment by jenya239** [comment]
Very interesting, but where is the documentation for functions like mrb_* (mrb_load_irep for example)?

**Reply by jenya239** [reply]
I mean except this https://mruby.org/docs/api/headers/mruby.h.html with very short descriptions

**Reply by katafrakt** [reply]
@jenya239 `mrb_load_irep` is defined in `irep.h`, not `mruby.h`, see: https://github.com/mruby/mruby/blob/61f57b22dd76c4d2f85721417aa21c2bbc4d3e11/include/mruby/irep.h#L88-L106

I'm mostly going through the header files themselves to get info.

**Reply by jenya239** [reply]
I'm surprised there is no separate documentation

**Reply by jenya239** [reply]
@katafrakt do you plan create more mruby guides?

**Reply by katafrakt** [reply]
@jenya239 not sure if guides, but I have a couple posts planned. I don't know when I will have a chance to make them happen.

**Reply by jenya239** [reply]
@katafrakt I can't reproduce https://katafrakt.me/2024/10/05/mruby-beyond-hello-world/. There is a problem with the jbreeden/mruby-sqlite. I am worried that it is very old. I installed sqlite3 and libsqlite3-dev. In compile time I got HUGE amount of warnings like 
```
warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
98192 |     AGGREGATE(sum,               1, 0, 0, sumStep,         sumFinalize    ),
```
and in runtime
`uninitialized constant now::SQLITE_READONLY_RECOVERY (NameError)`

**Reply by katafrakt** [reply]
@jenya239 I picked jbreeden/mruby-sqlite because it's self-contained - the code of SQLite is provided with the gem and it should not rely on you system sqlite3. I retraced the steps in the blog post and was able to successfully run it (set aside the issue with `mruby-print` gem being removed in a recent mruby release).

The error you provided suggests that the database opened in a read-only mode, for some reason. Maybe due to some filesystem setting? Or maybe you are already connected to this database somewhere?

I wouldn't pay too much attention to compilation warnings though. They are kind of expected.

**Reply by jenya239** [reply]
ok I'll check this again. Did you use mruby 3.3?

**Reply by katafrakt** [reply]
I've been using most recent version from Github, but yes, it was 3.3 version in both cases.
